integration/integrationtest/docker.go

```diff
@@ -403,6 +405,12 @@ func RunLocalDockerRegistry(t testing.TB, pool *dockertest.Pool, conf RegistryCo
 	})
 	require.NoError(t, err)
 
+	t.Cleanup(func() {
+		if !t.Failed() {
+			_ = pool.Purge(resource)
+		}
```

Comment:
Don't we always want to cleanup?

Reply:
It becomes very hard to debug if you do 😢